byron/ledger/executable-spec/test/Ledger/Delegation/Properties.hs

```diff
@@ -405,10 +411,65 @@ tracesAreClassified = traceLengthsAreClassified @DELEG 1000 100
 -- invalid signals are requested.
 invalidSignalsAreGenerated :: Property
 invalidSignalsAreGenerated = withTests 300 $ property $ do
+
   let aTraceLength = 100
       failureProfile = failures profile
+
   tr <- forAll (invalidTrace @DBLOCK aTraceLength failureProfile)
+
   cover 80
         "Invalid signals are generated when requested"
         (isLeft $ Invalid.Trace.errorOrLastState tr)
-  -- TODO: classify the kind of failures we get.
+
+  case Invalid.Trace.errorOrLastState tr of
+    Left pfs -> do
+      let
+        sdelegFailures :: [PredicateFailure SDELEG]
+        sdelegFailures = extractValues pfs
```

Comment:
This is a neat trick! 👏 Now I see why you left the `Data` & `Typeable` instances in 😉

Reply:
Where some see neatness some others see laziness :P But yeah, that was exactly my thought: "since we have the `Data` and `Typeable` instances, why don't we just use them?" :smile: